roles/openstack/neutron/tasks/cleanup.yml

```diff
@@ -0,0 +1,30 @@
+---
+# This file is used to cleanup OpenStack neutron resources
+
+- name: Cleanup routers
+  vars:
+      prefix: "{{ provisioner.prefix | default(uuid) }}-"
+  os_router:
+      interfaces: "{{ item.value.attach_subnets | map('add_prefix', prefix) | join(',') }}"
```

Comment:
Do you need `interfaces` and `network` for router cleanup? isn't router name enough?